File: domain/src/main/java/com/ortin/gesturetranslator/domain/repository/HandDetectionRepository.java
package com.ortin.gesturetranslator.domain.repository;

import androidx.annotation.NonNull;

import com.ortin.gesturetranslator.domain.listeners.DetectionHandListener;
import com.ortin.gesturetranslator.domain.models.Image;
import com.ortin.gesturetranslator.domain.models.SettingsMediaPipe;

public interface HandDetectionRepository {
    void detectImage(@NonNull Image image);

    void setDetectionHandListener(@NonNull DetectionHandListener detectionHandListener, @NonNull SettingsMediaPipe settingsMediaPipe);
}

File: domain/src/main/kotlin/com/ortin/gesturetranslator/domain/repository/HandDetectionRepository.kt
package com.ortin.gesturetranslator.domain.repository

import android.graphics.Bitmap
import com.ortin.gesturetranslator.domain.listeners.DetectionHandListener
import com.ortin.gesturetranslator.domain.models.ImageDetected
import com.ortin.gesturetranslator.domain.models.SettingsMediaPipe
import com.ortin.gesturetranslator.domain.models.VideoDetected
import com.ortin.gesturetranslator.domain.models.VideoFileDecode

interface HandDetectionRepository {
    fun detectImage(image: Bitmap): ImageDetected?
    suspend fun detectVideoFile(videoFile: VideoFileDecode): VideoDetected?
    fun detectLiveStream(image: Bitmap)
    suspend fun setSettingsModel(settingsMediaPipe: SettingsMediaPipe)
    fun setMPDetectionListener(detectionHandListener: DetectionHandListener)
}
